versioned/storage/common/src/test/java/org/projectnessie/versioned/storage/common/indexes/TestKeyIndexSets.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.storage.common.indexes;
+
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.projectnessie.versioned.storage.common.indexes.StoreIndexElement.indexElement;
+import static org.projectnessie.versioned.storage.common.indexes.StoreIndexes.deserializeStoreIndex;
+import static org.projectnessie.versioned.storage.common.indexes.StoreIndexes.newStoreIndex;
+import static org.projectnessie.versioned.storage.common.indexes.StoreKey.key;
+import static org.projectnessie.versioned.storage.common.objtypes.CommitOp.Action.ADD;
+import static org.projectnessie.versioned.storage.common.objtypes.CommitOp.COMMIT_OP_SERIALIZER;
+import static org.projectnessie.versioned.storage.common.objtypes.CommitOp.commitOp;
+import static org.projectnessie.versioned.storage.common.persist.ObjId.randomObjId;
+
+import java.util.stream.Stream;
+import org.assertj.core.api.SoftAssertions;
+import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
+import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.projectnessie.nessie.relocated.protobuf.ByteString;
+import org.projectnessie.versioned.storage.common.objtypes.CommitOp;
+import org.projectnessie.versioned.storage.commontests.ImmutableRealisticKeySet;
+import org.projectnessie.versioned.storage.commontests.KeyIndexTestSet;
+
+@ExtendWith(SoftAssertionsExtension.class)
+public class TestKeyIndexSets {
+  @InjectSoftAssertions SoftAssertions soft;
+
+  @Test
+  void foo() {
```

Comment:
Maybe rename? ;-) (or even remove fwiw – not sure if this is testing something fundamentally different from below)

Reply:
sure